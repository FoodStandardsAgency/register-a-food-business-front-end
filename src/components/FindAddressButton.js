import { Button } from "@slice-and-dice/govuk-react";
import { withTranslation } from "../../i18n.js";

const FindAddressButton = (props) => (
  <Button id="find-address" type="submit">
    {props.t("Find address")}
  </Button>
);

export default withTranslation("common")(FindAddressButton);
