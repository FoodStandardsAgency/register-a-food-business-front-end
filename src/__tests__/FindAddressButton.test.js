import FindAddressButton from "../components/FindAddressButton";
import { shallow, mount } from "enzyme";
import renderer from "react-test-renderer";
import * as emotion from "emotion";
import { createSerializer } from "jest-emotion";

expect.addSnapshotSerializer(createSerializer(emotion));

describe("<FindAddressButton />", () => {
  it("renders without crashing", () => {
    const wrapper = shallow(<FindAddressButton />);
    expect(wrapper.length).toBe(1);
  });

  it("matches the previous snapshot", () => {
    const tree = renderer.create(<FindAddressButton />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
