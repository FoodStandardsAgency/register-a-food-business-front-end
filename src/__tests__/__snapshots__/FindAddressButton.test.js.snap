// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FindAddressButton /> matches the previous snapshot 1`] = `
.emotion-1 {
  background-color: #00823b;
  border: none;
  box-shadow: 0 2px 0 #003618;
  color: #ffffff;
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  font-family: "nta",Arial,sans-serif;
  font-weight: 400;
  font-size: 1em;
  line-height: 1.25;
  outline-offset: -1px;
  outline: 1px solid transparent;
  padding: .526315em .789473em .263157em;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-appearance: none;
  -webkit-font-smoothing: antialiased;
  opacity: 1;
  pointer-events: auto;
  padding-right: .84211em;
  margin-bottom: 15px;
}

.emotion-1:hover {
  background-color: #00692f;
  color: #ffffff;
}

.emotion-1:focus {
  color: #ffffff;
  background-color: #00692f;
  outline: 3px solid #ffbf47;
}

.emotion-1:active {
  position: relative;
  top: 2px;
  box-shadow: 0 0 0 #003618;
}

.emotion-1:visited {
  color: #00823b;
}

.emotion-1 svg {
  max-width: 15px;
  margin-left: 20px;
}

@media only screen and (min-width:641px) {
  .emotion-1 {
    margin-bottom: 20px;
  }
}

<button
  className="emotion-0 emotion-1 emotion-2"
  disabled={false}
  id="find-address"
  type="submit"
>
  Find address
</button>
`;
